cpp/dolfinx/mesh/utils.h

```diff
@@ -888,6 +888,235 @@ Mesh<typename std::remove_reference_t<typename U::value_type>> create_mesh(
               std::move(geometry));
 }
 
+/// @brief Create a distributed mixed-topology mesh from mesh data using a
+/// provided graph partitioning function for determining the parallel
+/// distribution of the mesh.
+///
+/// From mesh input data that is distributed across processes, a
+/// distributed mesh::Mesh is created. If the partitioning function is
+/// not callable, i.e. it does not store a callable function, no
+/// re-distribution of cells is done.
+///
+/// @note This is an experimental specialised version of `create_mesh` for mixed
+/// topology meshes, and does not include cell reordering.
+///
+/// @param[in] comm Communicator to build the mesh on.
+/// @param[in] commt Communicator that the topology data (`cells`) is
+/// distributed on. This should be `MPI_COMM_NULL` for ranks that should
+/// not participate in computing the topology partitioning.
+/// @param[in] cells Cells on the calling process, as a list of lists,
+/// one list for each cell type. The cells are defined by their
+/// 'nodes' (using global indices) following the Basix ordering, and
+/// concatenated to form a flattened list. For lowest order cells this will be
+/// just the cell vertices. For higher-order cells, other cells 'nodes' will be
+/// included. See dolfinx::io::cells for examples of the Basix ordering.
+/// @param[in] elements Coordinate elements for the cells.
```

Comment:
Should this list be consistent across all processes? I.e. Even if a process only has triangular elements, and another quadrilaterals, both should send in (triangle, quad) coordinate element to ensure that everything is consistent at the point of code generation?

this would mean that some of the lists in cells could be empty (But that shouldn’t be a problem?)

Reply:
I can add a comment about that.